### ISSUE TITLE
Configurable region in AWS components

### ISSUE THREAD

**Issue by laardee** [serverless/components#219]
# Description

AWS components in registry are currently using hardcoded `us-east-1` region. Those should be configurable...

**Comment by laardee** [comment]
Should there be an option to deploy individual components to different regions? For example, if I have a service that I want to deploy to `eu-west-1`, but it has lambda@edge functions which need to be deployed to `us-east-1`.

**Comment by eahefnawy** [comment]
Hey @laardee, thanks for your PR! Do you think it should also be configurable from component inputs as well?

**Comment by laardee** [comment]
Yes, I think it should be configurable from the inputs. When are the inputs ready for the component? 

Because if inputs are not available when initializing the service object,  e.g. 
```js
const APIGateway = new AWS.APIGateway({ region: process.env.AWS_DEFAULT_REGION || 'us-east-1' })
``` 
it has to be initialized after `deploy` and `remove` functions are called.

Something like this

```js
const APIGateway = (region) => new AWS.APIGateway({ region: region || process.env.AWS_DEFAULT_REGION || 'us-east-1' })

const createApi = async (inputs) => {
  await APIGateway(inputs.region).createDeployment({......})
}

const deploy = async (inputs, context) => { 
  await createApi(inputs)
}
```
Or something else, that is not very pretty.

**Comment by saitai0802** [comment]
@eahefnawy Thank you for this awesome plugin! Would you please tell me if there are any updates for other regions at this moment(My choice is "**us-east-1**")? 

Also, would you suggest developers who would like to do something like the **retail-app component** to use the original way to build similar applications or just wait for you guys' update?

**Comment by saitai0802** [comment]
Would you please tell me will #220 setting overrides region setting like `webFrontend.hostingRegion` or `productsDb.inputs.region` in serverless.yml?
